fbw-common/src/wasm/systems/systems/src/hydraulic/command_sensor_unit.rs

```diff
@@ -0,0 +1,336 @@
+            (0b00001, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b10001) => CSU::OutOfDetent,
+            (0b10001, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b00011) => CSU::OutOfDetent,
+            (0b10001, 0b10001) => CSU::ConfFull,
+            _ => CSU::Fault,
+        }
+    }
+}
+
+pub struct CSUMonitor {
+    handle_position_id: VariableIdentifier,
+
+    current_switch_pattern: (SwitchPattern, SwitchPattern),
+    current_detent: CSU,
+    previous_switch_pattern: (SwitchPattern, SwitchPattern),
+    previous_detent: CSU,
+
+    last_valid_switch_pattern: (SwitchPattern, SwitchPattern),
+    time_since_last_valid_detent: Duration,
+}
+impl CSUMonitor {
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            handle_position_id: context.get_identifier("FLAPS_HANDLE_INDEX".to_owned()),
+            current_switch_pattern: (0b11000, 0b11000),
+            current_detent: CSU::Conf0,
+            previous_switch_pattern: (0b11000, 0b11000),
+            previous_detent: CSU::Conf0,
+            last_valid_switch_pattern: (0b11000, 0b11000),
+            time_since_last_valid_detent: Duration::ZERO,
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext) {
+        self.time_since_last_valid_detent += context.delta();
+    }
+
+    fn get_csu_configuration(&self) -> CSU {
+        CSU::get_csu_configuration(self.current_switch_pattern.0, self.current_switch_pattern.1)
+    }
+
+    pub fn get_last_valid_detent(&self) -> CSU {
+        CSU::get_csu_configuration(
+            self.last_valid_switch_pattern.0,
+            self.last_valid_switch_pattern.1,
+        )
+    }
+
+    pub fn time_since_last_valid_detent(&self) -> Duration {
+        self.time_since_last_valid_detent
+    }
+
+    pub fn get_current_detent(&self) -> CSU {
+        self.current_detent
+    }
+
+    pub fn get_previous_detent(&self) -> CSU {
+        self.previous_detent
+    }
+}
+impl SimulationElement for CSUMonitor {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        // The MSFS flap handle only delivers valid positions.
+        // TODO: it could possible to take into account `CSU::OutOfBounds` positions
+        // when using an axis binded to a flap lever or interfacing with a hardware CSU.
+        //
+        // The output is a tuple with identical values to simulate two parallel switches as per
+        // the real plane.
+        fn to_switch_pattern(value: u8) -> (SwitchPattern, SwitchPattern) {
+            match value {
+                0 => (0b11000, 0b11000),
+                1 => (0b01100, 0b01100),
+                2 => (0b00110, 0b00110),
+                3 => (0b00011, 0b00011),
+                4 => (0b10001, 0b10001),
+                _ => (0b00000, 0b00000), // Setting a value that leads to `CSU::Fault` to complete all
+                                         // the alternatives of the match statement.
+            }
+        }
```

Comment:
Why not define this function outside of this block?
Another option would be to inline it because it's only used once.